### DISCUSSION TITLE
glfw (or libxi) install requires inputproto >= 2.3.99.1 while I don't find how to update :(

### DISCUSSION THREAD

**Discussion by Extra-Creativity** [xmake-io/xmake#2672] [Q&A] [answered]
I'm trying to install glfw through `xrepo install glfw`. However, it prompts some error info: 

```
install libxi 1.8 .. failed

configure: WARNING: xmlto not found - documentation targets will be skipped
configure: WARNING: fop not found - documentation targets will be skipped
configure: WARNING: xsltproc not found - cannot transform XML documents
configure: WARNING: asciidoc not found - documentation targets will be skipped
configure: error: Package requirements (xproto >= 7.0.13 x11 >= 1.6 xextproto >= 7.0.3 xext >= 1.0.99.1 inputproto >= 2.3.99.1) were not met:
Requested 'inputproto >= 2.3.99.1' but version of InputProto is 2.3.2
Consider adjusting the PKG_CONFIG_PATH environment variable if you installed software in a non-standard prefix.
Alternatively, you may set the environment variables XI_CFLAGS and XI_LIBS to avoid the need to call pkg-config.
```

However, when I google for libxi/inputproto, I don't find any valid info to update the inputproto to 2.3.99.1 or newer. I have another project that doesn't use Xmake but glfw can work. I'm quite confused why this will happen!

My platform is Ubuntu20.04, if needed. Any advice would be appreciated!

**Comment by waruqi** [comment, marked as answer]
you can improve xmake-repo/libxi package or use apt install it `sudo apt install libxi-dev`

**Reply by Extra-Creativity** [reply]
Thanks! By the way, It seems that ImGui in xrepo doesn't support [backends] options, like imgui-[opengl3], while vcpkg provides them aside from core imgui (so that installing imgui and imgui-[vulkan] will make imgui work in vulkan). It would be nice if it's supported since imgui is often used in graphics engine or tools.

Backends code is [here](https://github.com/ocornut/imgui/tree/master/backends).

**Reply by waruqi** [reply]
you can enable glfw_opengl3 config

https://github.com/xmake-io/xmake-repo/blob/373b34851b03d85702c060d8b8959ef52d409dd0/packages/i/imgui/xmake.lua#L26

or open a pr to add other backend config
